### PR TITLE
implemented with two goroutines, process took 1.973694013s

### PR DESCRIPTION
**Pull request by mkuptsov**
implemented with two goroutines, process took 1.973694013s

### CODE REVIEW

File: 1-producer-consumer/main.go [hunk 1]
```diff
@@ -10,22 +10,23 @@ package main
 
 import (
 	"fmt"
+	"sync"
 	"time"
 )
 
-func producer(stream Stream) (tweets []*Tweet) {
+func producer(stream Stream, tweets chan Tweet) {
```
Comment (unspectd) [inline 1-producer-consumer/main.go:17]: chan *Tweet?

Comment (unspectd) [reply 1-producer-consumer/main.go:17]: Probably make it write-only?

File: 1-producer-consumer/main.go [hunk 2]
```diff
@@ -38,11 +39,21 @@ func main() {
...
-	tweets := producer(stream)
+	go func() {
+		defer wg.Done()
+		producer(stream, tweets)
+	}()
 
 	// Consumer
-	consumer(tweets)
-
+	go func() {
```
[the remark this answers is not in the document]
Comment (mkuptsov) [reply 1-producer-consumer/main.go:53]: yes, it isn't necessary, corrected.

File: 1-producer-consumer/main.go (hunk 1 above)
Comment (mkuptsov) [reply 1-producer-consumer/main.go:17]: corrected

**Pull request merged**